### PR TITLE
Add Li2 STO tests

### PR DESCRIPTION
**Pull request by jtkrogel**
Addresses #394.  VMC and DMC tests are provided for the Li dimer system used in Umrigar, Nightingale, & Runge's 1993 JCP.  This is a Slater (STO) Jastrow (Pade) wavefunction.  

The VMC energy matches UNR to 0.2 mHa.  A timestep study was performed to compare the Green's function in QMCPACK with UNR.  QMCPACK and UNR agree for moderately small timesteps, with a small difference (~1 mHa) apparently present in the t=0 limit.  Please read the README file included in this PR as part of the review.

The QMC_RUN_AND_CHECK function was modified to handle an arbitrary number of series checks (using the catch-all ARGN input list), so that e.g. VMC and DMC results generated from a single run can be checked without redoing the run.  This might be also be of use for future tests as a function of timestep.  Series checks added beyond the first result in a series number being added to the full test name; the first series test does not contain the series number in the name for backwards compatibility.

A final point worth noting is that v3.0.0, v3.1.1, and v3.2.0 (AoS) are all of a similar speed, while SVN revision 6259 is about 16% faster.

### CODE REVIEW

**Comment by markdewing** [comment]
okay to test

**Comment by jtkrogel** [comment]
I'm not sure of the minimum version, but this worked with CMake 2.8.10

File: CMake/macros.cmake [hunk 1]
```diff
@@ -290,7 +290,7 @@ ENDFUNCTION()
 # SHOULD_SUCCEED - whether the test is expected to pass or fail.  Expected failing tests will not have
 #                  the scalar tests added.
 
-FUNCTION(QMC_RUN_AND_CHECK BASE_NAME BASE_DIR PREFIX INPUT_FILE PROCS THREADS SCALAR_VALUES SERIES SHOULD_SUCCEED)
+FUNCTION(QMC_RUN_AND_CHECK BASE_NAME BASE_DIR PREFIX INPUT_FILE PROCS THREADS SHOULD_SUCCEED)
```
Comment (markdewing) [inline CMake/macros.cmake]: Could you update the documentation of the function arguments above?   Some changes in the order, plus the added optional argument pairs for series and scalar values.

File: tests/system/CMakeLists.txt [hunk 2]
```diff
@@ -326,6 +326,66 @@ ELSE()
...
+  LIST(APPEND LI2STOSJS_VMC_SCALARS "totenergy" "-14.947214  0.000502") # total energy
+  LIST(APPEND LI2STOSJS_VMC_SCALARS "variance"  "  0.170618  0.001035") # energy variance
+  LIST(APPEND LI2STOSJS_DMC_SCALARS "totenergy" "-14.989438  0.000643") # total energy
+  LIST(APPEND LI2STOSJS_DMC_SCALARS "variance"  "  0.151118  0.000312") # energy variance
+
+  QMC_RUN_AND_CHECK(short-li2_sto-sj_dmc
+                    "${CMAKE_SOURCE_DIR}/tests/molecules/Li2_STO_ae"
+                    Li2.STO.short
+                    Li2.STO.short.in.xml
+                    16 1
```
Comment (markdewing) [inline tests/system/CMakeLists.txt]: These tests are only added for MPI builds - is it preferable to have 16 procs and 1 thread or 1 proc and 16 threads (the latter would at least work on non-MPI builds)?

**Comment by prckent** [comment]
Good catch. An new / updated PR is needed. Unless we are seeking to test MPI functionality (which we do in other tests) 16 threads and 1 (MPI) process is preferred.

**Comment by markdewing** [comment]
The tests don't work with SOA builds.  But maybe we want to leave that failure in place to motivate fixing the code.

**Comment by prckent** [comment]
@markdewing Exactly. The SoA work is far from being done and this is an easy way to identify what does not yet work. The idea is to put tests for all part of the code that we intend to preserve moving forward. This is needed for SoA in the short term and Kokkos etc. in the long term.

**Comment by jtkrogel** [comment]
OK, I can put in a new PR switching to 1x16 and updating the docs.

Another set of fixes that might be worth putting in place are actual aborts when someone tries to use e.g. the complex/gpu code on gaussian/sto inputs and have tests catching the expected aborts.

**Comment by ye-luo** [comment]
LCAO with SoA is being worked. Please leave the tests always on.

**Comment by prckent** [comment]
Skip with GPU and Complex. Run with SoA.

**Pull request merged**